lib/semian/lru_hash.rb

```diff
@@ -83,16 +89,36 @@ def clear_unused_resources
     return unless @lock.try_lock
     # Clears resources that have not been used in the last 5 minutes.
     begin
+      timer_start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+
+      payload = {
+          size: @table.size,
+          examined: 0,
+          cleared: 0,
+          elapsed: nil,
+      }
+
+      stop_time = Time.now - MINIMUM_TIME_IN_LRU # Don't process resources updated after this time
       @table.each do |_, resource|
-        break if resource.updated_at + MINIMUM_TIME_IN_LRU > Time.now
+        payload[:examined] += 1
+
+        # The update times of the resources in the LRU are monotonically increasing,
+        # time, so we can stop looking once we find the first resource with an
+        # update time after the stop_time.
+        break if resource.updated_at > stop_time
+
+        # TODO(michaelkipper): Should this be a flag?
         next if resource.in_use?
 
         resource = @table.delete(resource.name)
         if resource
-          Semian.notify(:lru_hash_cleaned, self, :cleaning, :lru_hash)
```

Comment:
This would mean that anyone subscribing to this event would no longer get it. If we're okay with that, we should still call that out in a changelog.

Reply:
This PR is merging to a branch that only Core uses.

Reply:
Ah, I missed we were merging into a branch :+1: